app/src/main/java/com/example/myapplication/MainActivity.kt

```diff
@@ -1,19 +1,135 @@
+
+    var studentList = mutableListOf<Student>()
+    var disciplineList = mutableListOf<Discipline>()
+    var listOfMarks = mutableListOf<Mark>()
+    var averageMark: Int = 0
+
     override fun onCreate(savedInstanceState: Bundle?) {
+
+        // DONE 3
+        val student1 = Student("Ivan", 1)
+        val student2 = Student("Kate", 2)
+        val student3 = Student("Nick", 3)
+        val student4 = Student("Vera", 4)
+        val student5 = Student("Alisa", 5)
+
+        studentList = mutableListOf(student1, student2, student3, student4, student5)
+
+        val discipline1 = Discipline("Maths", 1)
+        val discipline2 = Discipline("Chemistry", 2)
+        val discipline3 = Discipline("English", 3)
+
+        disciplineList = mutableListOf(discipline1, discipline2, discipline3)
+
+        createRandomMarks()
+        listOfMarks.forEach { mark -> Log.i("TMS", "$mark") }
+        calculateAverageMarkForStudent()
+        calculateAverageMarkForDiscipline()
+        calculateAverageMarkForClass()
+        findGoodAndExcellentStudents()
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
+
+    // DONE 4
+    private fun createRandomMarks() {
+        studentList.forEach { student ->
+            disciplineList.forEach { discipline ->
+                listOfMarks.add(
+                    Mark(
+                        (1..10).random(),
+                        student,
+                        discipline
+                    )
+                )
+            }
+        }
+    }
+
+    // DONE 5-6
+    private fun calculateAverageMarkForStudent() {
+        for (i in 0 until studentList.size) {
+            listOfMarks.forEach { mark ->
+                if (mark.student.studentId == i + 1) {
+                    averageMark += mark.value
+                }
+            }
+            Log.v("TMS", "Cредний балл ученика id${i + 1}: ${averageMark / disciplineList.size}")
+            averageMark = 0
+        }
+    }
+
+    private fun calculateAverageMarkForDiscipline() {
+        for (i in 0 until disciplineList.size) {
+            listOfMarks.forEach { mark ->
+                if (mark.discipline.disciplineId == i + 1) {
+                    averageMark += mark.value
+                }
+            }
+            Log.w("TMS", "Cредний балл предмета id${i + 1}: ${averageMark / studentList.size}")
+            averageMark = 0
+        }
+    }
+
+    private fun calculateAverageMarkForClass() {
+        listOfMarks.forEach { mark ->
+            averageMark += mark.value
+        }
+        Log.w("TMS", "Cредний балл класса: ${averageMark / listOfMarks.size}")
+    }
+
+    // DONE 9*
+    private fun findGoodAndExcellentStudents() {
+        var counterForExcellentStudents = 0
+        var counterForGoodStudents = 0
+        for (i in 0 until studentList.size) {
+            listOfMarks.forEach {
+                if (it.student.studentId == i + 1) {
```

Comment:
Such check works only when Student list is immutable and ids - are the following progression
 Works fine for current case. But for general, it's better compare exact id.
'''
studentList.forEach { student ->
  listOfMarks.forEach { mark ->
    if (mark.student.id ==student.id)
...
  }
}